### PR TITLE
Indirect - Diffraction Reduction - Correctly implement Use Manual Grouping

### LINKED ISSUE #20477: Indirect - Diffraction Reduction - Use Individual Grouping not functional
The Use Individual Grouping option in the Indirect Diffraction Reduction interface, should behave similarly to the 'Groups' Detector Grouping mode in the Indirect > Data Reduction > ISIS Energy Transfer interface.

Currently, this option doesn't have any function.

### PR DESCRIPTION
**Pull request by brandonhewer**
In the Indirect Diffraction Reduction interface, the 'Use Individual Grouping' option does not work as intended and currently has no function in the interface. This option should behave similarly to the 'Groups' mode of the the Detector Grouping option in ISISEnergyTransfer.

This PR ensures the Use Individual Grouping option (now the Use Manual Grouping option, in order to avoid confusion with Individual Grouping mode in ISISEnergyTransfer)  works as intended.

**To test:**
1. Ensure that you have 'Use Data Archive' enabled.
2. Navigate to Interfaces > Indirect > Diffraction.
3. Enter 26176 into the Sample field.
4. Check the checkbox labelled 'Use Manual Grouping' and ensure the spinner labelled 'Number of Groups' is enabled.
5. Select any number of groups > 0 and note the number of groups selected.
6. Run the Algorithm.
7. Open the output workspace and ensure the number of spectra is equal to the number of selected groups (spectra has been summed into N groups).

Fixes #20477 

**Release Notes** 

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Inelastic/IndirectReductionCommon.py [hunk 1]
```diff
@@ -487,24 +503,25 @@ def group_spectra(workspace_name, masked_detectors, method, group_file=None, gro
 
         # Mask detectors if required
         if len(masked_detectors) > 0:
-            MaskDetectors(Workspace=workspace_name,
-                          WorkspaceIndexList=masked_detectors)
+            mask_detectors = AlgorithmManager.create("MaskDetectors")
+            mask_detectors.setChild(True)
+            mask_detectors.setProperty("Workspace", workspace)
+            mask_detectors.setProperty("WorkspaceIndexList", masked_detectors)
+            mask_detectors.execute()
```
Comment (edwardb96) [inline scripts/Inelastic/IndirectReductionCommon.py]: Could this be moved to it's own function?

File: scripts/Inelastic/IndirectReductionCommon.py [hunk 2]
```diff
@@ -693,11 +710,11 @@ def save_reduction(workspace_names, formats, x_units='DeltaE'):
...
-            saveAsciiAlg = AlgorithmManager.createUnmanaged('SaveAscii', 2)
-            saveAsciiAlg.initialize()
-            saveAsciiAlg.setProperty('InputWorkspace', workspace_name)
-            saveAsciiAlg.setProperty('Filename', workspace_name + '.dat')
-            saveAsciiAlg.execute()
+            save_ascii_alg = AlgorithmManager.createUnmanaged('SaveAscii', 2)
+            save_ascii_alg.initialize()
+            save_ascii_alg.setProperty('InputWorkspace', workspace_name)
+            save_ascii_alg.setProperty('Filename', workspace_name + '.dat')
+            save_ascii_alg.execute()
```
Comment (edwardb96) [inline scripts/Inelastic/IndirectReductionCommon.py]: Same again, could this be its own function?

**Review by edwardb96 — CHANGES_REQUESTED** [review]
Now working 👍. Just a couple of potential method extractions...

**Comment by brandonhewer** [comment]
@edwardb96 Good spots, I've moved the highlighted code to separate functions. Should help in slowly cleaning up IndirectReductionCommon.

**Review by edwardb96 — APPROVED** [review]

**Comment by edwardb96** [comment]
Ok to merge when tests pass. 😄

**Comment by AndreiSavici** [comment]
Is this supposed to be documented somewhere? Release notes?

**Comment by martyngigg** [comment]
I'm going to me this so it can be tested by scientists
@brandonhewer Please add something to the inelastic release notes in a separate PR.

**Pull request merged**